README.md

```diff
@@ -1,119 +1,180 @@
+
+Reason we are replacing the driver is that it doesn't allow us to set PDM frequency that is required by the microphone (see datasheet) and we need to use PDM CLK pin as a GPIO for setting AAD modes.
+
+T5838 datasheet for more information can be found [here](https://invensense.tdk.com/wp-content/uploads/2023/06/DS-000383-T5838-Datasheet-v1.1.pdf).
 
 ## Setup
 
-If you do not already have them you will need to:
+1. To your `west.yml` add the IRNAS remote to the `remotes` section:
 
-- [install west](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/gs_installing.html#install-west)
-- [install east](https://github.com/IRNAS/irnas-east-software)
+   ```yaml
+   - name: irnas
+     url-base: https://github.com/irnas
+   ```
 
-Then follow these steps:
+2. Then in the `projects` section add at the bottom:
 
-```shell
-west init -m https://github.com/IRNAS/<repo-name> <repo-name>
-cd <repo-name>/project
+   ```yaml
+   - name: irnas-t5838-driver
+      repo-path: irnas-t5838-driver
+      path: irnas/irnas-t5838-driver
+      remote: irnas
+      revision: feature/t5838-driver
+   ```
 
-# set up east globally (this only needs to be done once on each machine)
-east sys-setup
-# install toolchain for the version of NCS used in this project
-east update toolchain
+3. Add the flash DTS entry to your board definition or overlay file. For example:
 
-# set up west modules in the repository
-east bypass -- west update
-```
+   ```dts
+   dmic_dev: &pdm0 {
+      status = "okay";
+      pinctrl-0 = <&pdm0_default>;
+      pinctrl-1 = <&pdm0_sleep>;
+      pinctrl-names = "default","sleep";
+      #address-cells = < 0x1 >;
+      #size-cells = < 0x0 >;
+      t5838: t5838@0 {
+         status = "okay";
+         compatible = "tdk,t5838-nrf-pdm";
+         micen-gpios = <&gpio1 11 GPIO_ACTIVE_HIGH>;
+         thsel-gpios = <&gpio1 4 GPIO_ACTIVE_HIGH>;
+         wake-gpios = <&gpio1 10 GPIO_ACTIVE_HIGH>;
+         pdmclk-gpios = <&gpio1 13 GPIO_ACTIVE_HIGH>;
+         reg = <0x0>;
+      };
+   };
+   ```
 
-## Building and flashing
+   Note that since we are using dts entry pdm0 as a bus, we need to add the following to its definition:
 
-To build the application firmware:
+   ```dts
+   #address-cells = < 0x1 >;
+   #size-cells = < 0x0 >;
+   ```
 
-```bash
-cd app
-east build -b <board_name> -u <build_type>
-```
+   This tells the kernel that the bus has 1 address cell and 0 size cells. So we don't have to set reg value to full address of the device, but only the address of the device on the bus. Note that reg property is not currently used, as we currently only support one device on the bus and microphone proprietary protocol (I refer to it as fake2c communication) address is hardcoded in the driver header file (0x53 which is the only address microphone supports).
+   It is planed reg value will be used in the future to support two microphones on the same bus (stereo configuration).
```

Comment:
Review this line, if it is correct, I don't fully understand it.

Reply:
I do need to further review this, the thing about these two DT parameters is that DTS expects full 4byte reg value if we don't set that for any sub peripheral. I will look further into this.